cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -79,7 +79,7 @@ func StartTestServer(t *testing.T) (result *restclient.Config, tearDownForCaller
 
 	s := options.NewServerRunOptions()
 	s.InsecureServing.BindPort = 0
-	s.SecureServing.BindPort = freePort()
+	s.SecureServing.BindPort = 0
```

Comment:
We have a problem here: `0` is reserved for "do not listen". Can we have an utility function `CreateListenerOnFreePort(addr)` and use that here?

Reply:
Sure